### PR TITLE
More nix support and type insertion

### PR DESCRIPTION
**Pull request by mckeankylej**
This pr adds two features that greatly enhance usability.
* Now `dante-repl-command-line-default-methods` will select nix if `default.nix` or `shell.nix` is found in the root directory. Many people myself included have `default.nix` files that both `nix-build` and `nix-shell` removing the need for a `shell.nix`.
* Now `dante-type-at` will insert the type at the line above if you pass it an argument like so `(dante-type-at :insert)`

Also I am adding dante support to [spacemacs](http://spacemacs.org/). You can view my progress in this [repo](https://github.com/mckeankylej/spacemacs). To try it out yourself use my spacemacs repository and add the following to your `dotspacemacs-configuration-layers`:
```elisp
(haskell :variables haskell-completion-backend 'dante)
```
If you merge this pr I can pr my spacemacs repo into master and many many more people can use dante!

### CODE REVIEW

**Comment by jyp** [comment]
Thanks for this PR.

It could be good if you can document why you need the "insert" argument. As it stands it looks like dead code and is liable to be removed in an upcoming refactoring.

**Comment by mckeankylej** [comment]
Please don't remove it. The insert argument changes the behavior of the type-at function.
If you have the following:
```haskell
test = "hello"
```
And you hover over the symbol test and run 
```lisp
(dante-type-at :insert)
```
You will get
```haskell
test :: String
test = "hello"
```
If you leave the argument off the function has the same behavior as before the pr.

File: dante.el [hunk 1]
```diff
@@ -104,13 +104,14 @@ look for a .cabal file, or use the current dir)."
   (or dante-project-root
       (file-name-directory (or (dante-cabal-find-file) (dante-buffer-file-name)))))
 
-(defun dante-repl-by-file (root file cmdline)
-  (when (file-exists-p (concat root file)) cmdline))
+(defun dante-repl-by-file (root files cmdline)
+  (some (lambda (file) (when (file-exists-p (concat root file)) cmdline)) files))
```
Comment (michalrus) [inline dante.el]: @mckeankylej @jyp what is this `(some`?

```
flycheck-buffer: Symbol’s function definition is void: some
```

**Comment by mckeankylej** [comment]
Whoops it's a macro where it loops through a list and returns the first item that makes the predicate true. I will submit a patch that brings it in scope. I have it in scope in my config. Sorry about that this is my first use of lisp.

**Comment by michalrus** [comment]
@mckeankylej, look at https://github.com/jyp/dante/issues/11

You should not use `(some)` in library code, please, use `(cl-some)`. It will be in scope.

**Comment by puffnfresh** [comment]
@mckeankylej have you got a PR for Spacemacs yet?

**Comment by mckeankylej** [comment]
@puffnfresh I have been at a conference on the plane ride home I will pr

**Comment by mckeankylej** [comment]
@puffnfresh https://github.com/syl20bnr/spacemacs/pull/8697

**Pull request merged**